### DISCUSSION TITLE
FR: Apps routing to multiple output devices

### DISCUSSION THREAD

**Discussion by tomber42** [File-New-Project/EarTrumpet#634] [Ideas]
Hi, dont know if that is possible but it would be nice to routing an app to more than one output devices and can adjust it seperate.

**Comment by HaArD2000** [comment]
This would be great, I have several monitors and a soundbar and would love it if I could direct sound from VLC/Plex to all of them at the same time and tweak the volume of each device independently to get the right mix.

**Comment by SirNahka** [comment]
Thirded. Any idea when this might potentially be added? Is this in progress or anything like that?

**Reply by SmilerRyan** [reply]
I am also interested in this being added, For me, it would be great for streaming applications to a virtual audio cable as well as my speakers, so I don't have to direct my microphone to it and listen to myself as well.

**Comment by vavavr00m** [comment]
I agree with the OP and upvoted this. I own both Dante Via & Voicemeeter Potato but still looking for an open source alternative that is a blend of easy to understand/use, with an intuitive GUI, has unlimited audio sources & output destinations, able to do complex things (equalize, add multiple effects that can be easily shared akin to [Voicemod](https://www.voicemod.net/)) and, of course, that works well.

Here are some applications that can route audio:

- [Dante Via](https://www.audinate.com/products/software/dante-via) (Commercial) - Drag and drop, however, price is steep.

![image](https://user-images.githubusercontent.com/4675132/224713638-24fe9e54-51ac-46ec-82ab-f311de6c8991.png)

- Voicemeeter ([Regular](https://vb-audio.com/Voicemeeter/index.htm)/[Banana](https://vb-audio.com/Voicemeeter/banana.htm)/[Potato](https://vb-audio.com/Voicemeeter/potato.htm)) (Donationware/Nagware) - Not easy to understand for beginners, limited input and output, annoying donation nag

![image](https://user-images.githubusercontent.com/4675132/224717337-86826cd5-2dc5-4f89-a627-85c4d3fc0e37.png)

- [VB-Audio Matrix](https://vb-audio.com/Matrix/index.htm)* - Touted as a Universal Audio Router with 552 x 552 Points Real Time Audio Matrix for Windows. Currently in beta phase and is being developed by the same guys who develop Voicemeeter Potato.

![image](https://github.com/File-New-Project/EarTrumpet/assets/4675132/3d1b37c3-fcb8-478a-b107-5bcdc3b60506)

- [DS WASAPI ASIO Router Mixer](https://www.3delite.hu/Object%20Pascal%20Developer%20Resources/DSWASAPIASIORouterMixer.html)* 

![image](https://github.com/File-New-Project/EarTrumpet/assets/4675132/bdcfe51e-a736-4fe2-a0a8-2526d3d176c0)

- [Virtual Audio Cable](https://vac.muzychenko.net/en/) (Commercial) - doable but with some [work](https://www.raymond.cc/blog/virtual-audio-cable-enables-multiple-audio-outputs-in-windows-7/)

- [JACK Audio Connection Kit](https://jackaudio.org/applications/) (Open Source) - last time I tried, couldn't figure out how to make it work.

- [Audio Router](https://github.com/audiorouterdev/audio-router) (Open Source) - last commit/release was in 2016 but still works for simple routes by duplicating source

![image](https://user-images.githubusercontent.com/4675132/224723012-8f8c5cec-fb27-4210-9d24-196dcaca79d2.png)

- [VB-Audio-Router](https://github.com/ShortDevelopment/VB-Audio-Router)* (Open Source) - description says single app capture but there's no option in the GUI, multiple outputs is in preview mode but selection doesn't stick and no idea how to add multiple outputs.

![image](https://user-images.githubusercontent.com/4675132/224724679-2129a7e0-d437-48e7-8476-e9d40705da5f.png)

- [SynchronousAudioRouter](https://github.com/eiz/SynchronousAudioRouter)* (Open Source) - last release was in 2018, have not tried

![image](https://user-images.githubusercontent.com/4675132/224724393-55d16bb8-6f3b-416a-9a99-9f75c346c1c3.png)

- [PulseAudio](https://www.freedesktop.org/wiki/Software/PulseAudio/) (Open Source) - has some [function issues](https://www.freedesktop.org/wiki/Software/PulseAudio/Ports/Windows/Support/) with the Windows port, and also, it did not work for me on W10

- [Audio Repeater Pro](https://www.audiorepeater.com/) (Commercial) - have not tried.

![image](https://user-images.githubusercontent.com/4675132/227430897-23435d89-5993-4769-b921-c588dd4ac9c9.png)

- [Blue Cat's Connector](https://www.bluecataudio.com/Products/Product_Connector/)* - have not tried. Plugin-based (Available in VST, VST3, AAX or Audio Unit format). Seems okay though same steep pricing as Dante Via.

![image](https://user-images.githubusercontent.com/4675132/224777901-aa861d6b-96e0-47ea-a0fc-de10634cf6b5.png)

Other OS:

- [Loopback](https://rogueamoeba.com/loopback/)* (Commercial) - seems easy to use based on the screenshot provided

![image](https://user-images.githubusercontent.com/4675132/224724924-3a908431-3ecb-4ac0-858e-ca1ef3bd7c1e.png)

- [Audiomovers Omnibus](https://audiomovers.com/omnibus/)* (Commercial)

![image](https://github.com/File-New-Project/EarTrumpet/assets/4675132/0e730e9c-52da-4b31-9de6-577fda33f129)

- [SoundDesk](https://www.loudlab-app.com/sounddesk-app)* (Commercial)

- [QjackCtl](https://qjackctl.sourceforge.io/)* (Open Source) - JACK GUI

![image](https://user-images.githubusercontent.com/4675132/224723627-5b6a330c-f40f-49a7-bc45-22e3c547c83d.png)

- [PulseAudio Graph Control](https://github.com/futpib/pagraphcontrol)* (Open Source) - PulseAudio GUI

![image](https://user-images.githubusercontent.com/4675132/224723776-8624ef7b-d9c2-4342-baa0-6f263162e64a.png)

*Image taken from their official page

Chrome extensions:

- [AudioXout](https://chrome.google.com/webstore/detail/audioxout/eajkfkgkmngjbpckhidaichjnjiljbbl/)
- [Audio Pipes](https://chrome.google.com/webstore/detail/audio-pipes/ebmfdpppokkplmndpamkhbmochcdlefp)
- [AudioPick](https://chrome.google.com/webstore/detail/audiopick/gfhcppdamigjkficnjnhmnljljhagaha)

**Comment by Her0es** [comment]
Upping this. Would be a godsend to have this feature.
